dhall-docs/src/Dhall/Docs/Core.hs

```diff
@@ -230,6 +252,109 @@ getAllDhallFiles = emitErrors . map toDhallFile . foldr validFiles [] . filter h
         fromAssertion (Assert e) =  Just e
         fromAssertion _ = Nothing
 
+    {-
+    Validates the comments from source code against the `dhall-docs`
+    comment's specification
+
+    Non `dhall-docs` comments will be ignored, but invalid `dhall-docs`,
+    in addition, will log a warning.
+    -}
+    stripCommentSyntax :: Path Rel File -> Text -> GeneratedDocs (Maybe Text)
```

Comment:
Could the return type alternatively be `Either DocsGenWarning Text`?

I was thinking about this very nice blog post: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

I'm not yet sure how to best apply its advice here though…

Reply:
I've just read that, really nice! I chose `GeneratedDocs (Maybe Text)` to log `DocsGenWarning` easier. Also, I used `Maybe Text` on `FileComments` since not all files have headers.

I believe that the best application of that blog post's advice is to change the return type of `Dhall.Docs.Comment.parseComment` be `Either CustomParseError DhallComment`. I can give it a try.